ports.js

```diff
@@ -1,51 +1,228 @@
 /* eslint-disable no-console */
 /* globals encrypt:false, decrypt:false */
+const CONTENT_KEY = "pad";
 
-let app;
-if (typeof Elm === 'undefined') {
-  // This happens if we're in the context of Electron.
-  const Elm = require('./hoverpad.js');
-  app = Elm.Main.fullscreen();
-} else {
-  app = Elm.Main.fullscreen();
+const IS_WEB_EXTENSION = (typeof chrome === "object" && typeof chrome.storage === "object");
+
+function storePassphrase(passphrase) {
+  // Insecurely storing the passphrase.
+  sessionStorage.setItem("temporaryPassphrase", btoa(passphrase));
+  return Promise.resolve(null);
+}
+
+function dropPassphrase() {
+  sessionStorage.removeItem("temporaryPassphrase");
```

Comment:
Did you check this works in a webext context? I suppose so, but we're never too sure.